canonical-web-and-design/maas-ui
# feat(ui): add tab strip to machine details

## Done

- Added tab strip to machine details page.
- Added MachineDetails type, which extends the base Machine type. I've only included what I needed to get this branch to work with TS, but I've added a card (https://github.com/canonical-web-and-design/maas-ui/issues/1803) to type the rest. 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Click on a machine in the machine list and check that you are taken to /machine/<system_id>/summary
- Check that a tab strip shows, and that navigating to each tab correctly updates the selected state for the tab

## Fixes

Fixes canonical-web-and-design/MAAS-squad#3106

## Screenshot

![Screenshot_2020-10-22 casual-emu maas details bolla MAAS](https://user-images.githubusercontent.com/25733845/96833753-49b88480-1484-11eb-92a0-5008e6074a1a.png)
